### PR TITLE
cli: add follow flag to workflow execution

### LINKED ISSUE #295: cli: enable following workflow execution until finalisation
Implement a new flag to follow workflow execution (`--follow/-f`) after `reana-client start` or `reana-client run` until the workflow finalises:

```console
$ reana-client start --follow
...
...
$ # or
$ reana-client run --follow
...
...
```

Note that the exit code should match with the final status of the workflow so this is understood by any shell, and therefore, any CI runner would fail when the workflow fails.

### PR DESCRIPTION
**Pull request by leticiawanderley**
Closes #295

Adds a flag the start and run workflow commands to follow the
workflow execution until termination.

Signed-off-by: leticia <leticia.farias.wanderley@cern.ch>

### CODE REVIEW

File: reana_client/cli/workflow.py [hunk 1]
```diff
@@ -272,9 +276,14 @@ def workflow_create(ctx, file, name,
          'E.g. CACHE=off. (workflow engine - serial) '
          'E.g. --debug (workflow engine - cwl)',
 )
+@click.option(
+    '--follow', 'follow',
+    is_flag=True,
+    help='If set, follows the execution of the workflow until termination.',
```
Comment (tiborsimko) [inline reana_client/cli/workflow.py]: `default=False`

File: reana_client/cli/workflow.py [hunk 2]
```diff
@@ -43,6 +44,9 @@
 from reana_commons.utils import click_table_printer
 
 
+TIMECHECK = 5
```
Comment (tiborsimko) [inline reana_client/cli/workflow.py]: Would be good to move to the central `config.py` file.

File: reana_client/cli/workflow.py [hunk 3]
```diff
@@ -272,9 +276,14 @@ def workflow_create(ctx, file, name,
...
 )
+@click.option(
+    '--follow', 'follow',
+    is_flag=True,
+    help='If set, follows the execution of the workflow until termination.',
+)
 @click.pass_context
 def workflow_start(ctx, workflow, access_token,
-                   parameters, options):  # noqa: D301
+                   parameters, options, follow):  # noqa: D301
```
Comment (tiborsimko) [inline reana_client/cli/workflow.py]: This is not related to the PR, but perhaps you can fix a cosmetic help issue alongside this PR?  The help for `start` does not display well currently:

```console
$ reana-client start --help  
Usage: reana-client start [OPTIONS]

  Start previously created workflow.

  The `start` command allows to start previously created workflow. The
  workflow execution can be further influenced by passing input prameters
  using `-p` or `--parameters` flag and by setting additional operational
  options using `-o` or `--options`.  The input parameters and operational
  options can be repetitive. For example, to disable caching for the Serial
  workflow engine, you can set `-o CACHE=off`.

  Examples: \n \t $ reana-client start -w myanalysis.42 -p sleeptime=10 \n
  \t $ reana-client start -w myanalysis.42 -p myparam1=myvalue1 -p
  myparam2=myvalue2 -o CACHE=off
```

Note `\n` and `\t` in the ouput.  

Should look like: 

```console
$ reana-client run --help        
Usage: reana-client run [OPTIONS] SOURCES

  Shortcut to create, upload, start a new workflow.

  The `run` command allows to create a new workflow, upload its input files
  and start it in one command.

  Examples:

       $ reana-client run -n myanalysis-test-small -p myparam=mysmallvalue

       $ reana-client run -n myanalysis-test-big -p myparam=mybigvalue

Options:
  -f, --file PATH          REANA specifications file describing the workflow
                           and context which REANA should execute.
  -n, --name TEXT          Name of the workflow.

```

Comment (leticiawanderley) [reply reana_client/cli/workflow.py]: No problem, I found the issue.
I'll commit the changes now.

**Pull request merged**